ecommerce/EarningController.php

```diff
@@ -0,0 +1,292 @@
+<?php
+/**
+ * Manage earnings
+ *
+ * @package Tutor\Ecommerce
+ * @author Themeum <support@themeum.com>
+ * @link https://themeum.com
+ * @since 3.0.0
+ */
+
+namespace Tutor\Ecommerce;
+
+use Tutor\Helpers\QueryHelper;
+use Tutor\Models\OrderModel;
+use TUTOR\Singleton;
+
+/**
+ * Manage earnings
+ */
+class EarningController extends Singleton {
+
+	/**
+	 * Error message for the invalid earning data
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var string
+	 */
+	const INVALID_DATA_MSG = 'Invalid earning data';
+
+	/**
+	 * EarningController instance
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var EarningController object
+	 */
+	public static $instance = null;
+
+	/**
+	 * Earning table name
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var string
+	 */
+	private $earning_table;
+
+	/**
+	 * Order id
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var int
+	 */
+	private $order_id;
+
+	/**
+	 * Keep earning data here
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	public $earning_data = array();
+
+	/**
+	 * Instance of this class
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return EarningController object
+	 */
+	public static function get_instance() {
```

Comment:
No need to override `get_instance` method and property `public static $instance = null;`

```php
protected function __construct() {
  global $wpdb;
  $this->earning_table = $wpdb->prefix . 'tutor_earnings';
}
```